yal/utils.py

```diff
@@ -76,11 +77,22 @@ def normalise_phonenumber(phonenumber):
         raise ValueError("Invalid phone number")
 
 
+def extract_first_emoji(persona_emoji):
+    if is_emoji(persona_emoji):
+        return persona_emoji
+    emojis_preset = emoji_list(persona_emoji)
+    if emojis_preset:
+        return emojis_preset[0]["emoji"]
+    return ""
+
+
 def replace_persona_fields(text, metadata={}):
     for key in PERSONA_FIELDS:
         value = metadata.get(key)
         if value and value.lower() != "skip":
-            text = text.replace(f"[{key}]", value)
+            if key == "persona_emoji":
+                value = extract_first_emoji(value)
+            text = text.replace(f"[{key}]", "".join(value.splitlines()).strip())
```

Comment:
Are we trying to replace newlines with spaces using `splitlines`? What about `re.sub(r"\s+", " ", value)`, which will also replace any multiple spaces with a single space, and multiple newlines with a single space?